examples/tracking_pycaret.py

```diff
@@ -0,0 +1,218 @@
+                VALUES (?, ?, ?, ?, ?, ?)""",  # noqa: S608
+                list(chunk.itertuples(index=False, name=None)),
+            )
+        cursor.close()
+
+
+def read_data(table_name: str) -> pd.DataFrame:
+    """
+    Read data from database into pandas DataFrame.
+    """
+
+    query = f"""
+            SELECT
+                DATE_TRUNC('month', date) as month,
+                SUM(total_sales) AS total_sales
+            FROM {table_name}
+            GROUP BY month
+            ORDER BY month
+        """
+    with connect_database() as conn:
+        data = pd.read_sql(query, conn)
+
+    data["month"] = pd.to_datetime(data["month"], unit="ms")
+    data.sort_values(by=["month"], inplace=True)
+    return data
+
+
+def run_experiment(data: pd.DataFrame):
+    """
+    Run experiment on DataFrame, using PyCaret. Track it using MLflow.
+    The mlflow tracking is automatically executed by PyCaret.
+    """
+
+    # creating a blend of 3 models, which perform best on MASE metric
+    pycaret_setup = setup(data,
+                          fh=15,
+                          target="total_sales",
+                          index="month",
+                          log_experiment=True,
+                          verbose=False)
+
+    best3 = compare_models(sort="MASE", n_select=3)
+    tuned_models = [tune_model(i) for i in best3]
+    blended = blend_models(estimator_list=tuned_models, optimize="MASE")
+    best_model = finalize_model(blended)
+
+    # saving the model to disk
+    if not os.path.exists("model"):
+        os.makedirs("model")
+    save_model(best_model, 'model/crate-salesforecast')
+
+    # Create a name for the model
+    timestamp = int(time.time())
+
+    # registering the model with mlflow, but only if MLFLOW_TRACKING_URI is
+    # set to a tracking server
+    if not get_tracking_uri().startswith("file://"):
+        y_pred = best_model.predict()
+        signature = infer_signature(None, y_pred)
+        log_model(
+            sk_model=best_model,
+            artifact_path="crate-salesforecast",
+            signature=signature,
+            registered_model_name=f"crate-salesforecast-model-{timestamp}",
+        )
+    else:
+        print(# noqa: T201
+            "MLFLOW_TRACKING_URI is not set to a tracking server, "
+            "so the model will not be registered with mlflow")
+
+
+def main():
+    """
+    Provision dataset, and run experiment.
+    """
+
+    # Table name where the actual data is stored.
+    data_table = "sales_data_for_forecast"
+
+    # Provision data to operate on, only once.
+    if not table_exists(data_table):
+        import_data(data_table)
+
+        # Wait until table is ready.
+        i = 0
+        while not data_available(data_table) and i < 5:
+            i += 1
+            time.sleep(0.2)
+        if i == 5 and not data_available(data_table):
+            raise Exception("Data is not available in database table.")
```

Comment:
Well. That looks a bit inefficient. Which problem does this code aim to solve?

Reply:
Well this wasn't meant to be committed. Will submit a fix.

Reply:
Removed and replaced with something more  reasonable.
The reason for why I want to have something between provisioning and querying is to prevent bugs due to crate just being eventual consistent.
 
Just for consideration: I think in the other example, import_data() with directly following read_data() can - depending on the crate setup - lead to issues, if the refresh_interval of crate is higher than the time python needs to execute these two lines. This is a common encounter on my journeys, not only with crate (where on my 2008 notebook, everything works fine, but on my M2 Macbook, python is "just too fast", so the just inserted data are not available for querying yet).
During provisioning we should make sure, the database reached consistent state. What do you think?

Reply:
Most probably, `REFRESH TABLE ...` is your friend?

Reply:
Yeah that's what I'd suggest here. But as the other example was not created by me, I don't want to overrule the creator of the example. Just wanted to suggest, that I proably should also be fixed there.

Reply:
Hi. I am not sure to which example you are referring to here. Can you share a relevant link?

Reply:
https://github.com/crate-workbench/mlflow-cratedb/blob/0718a1283035c815596bdb31289ed0b6c10d8ff5/examples/tracking_merlion.py

Reply:
Apologies, I can't spot that loop there. Can you support me by referring to the exact line numbers.

Reply:
Ah, `REFRESH TABLE ...` is missing over there, right? Thanks! /cc @hammerhead

Reply:
> Apologies, I can't spot that loop there. Can you support me by referring to the exact line numbers?

The loop was committed by me on mistake. 
To set the record straight, it's a hacky vscode snippet of mine locally to quickly hack around these issues you get with data systems which are not strongly consistent - but shouldn't find it's way into any checked in code. So let's forget the loop once and for all ;-)
I'm suggesting to use refresh in the linked example, not the loop.

Reply:
Thanks. Deferred to GH-49.